x-pack/plugins/lens/public/xy_visualization/xy_config_panel/reference_line_config_panel/reference_line_panel.tsx

```diff
@@ -50,7 +51,7 @@ export const ReferenceLinePanel = (
   );
 
   const setConfig = useCallback(
-    (yConfig: Partial<YConfig> | undefined) => {
+    (yConfig: Partial<ExtendedYConfig> | undefined) => {
```

Comment:
Thanks for this idea. I see the other point here. But it seems to me, the `ReferenceLineConfig` is not reflecting, what it does too 😄. Before, it was `yConfig` with the arguments of the `extendedYConfig`. And, I believe, we can reuse the same structure for the `annotationLayer` expression. So, it will be the configuration of the elements, according to the `y` axis. 

What if we'll name it `yLineConfig` and `xLineConfig` expressions? We'll use one for `referenceLineLayer`, and the other - for `annotationLayer`. WDYT?

Reply:
Huh, but this is a bit more complex - one of the reason why yConfig name doesn't match is because reference line can be put on number interval chart on x axis too so that wouldn't match😅. Regarding annotation layer structure, it's becoming more and more different from this because we keep all information inside the `annotations[]` - and also we will have to rethink annotations anyway because when adding query annotations so I wouldn't touch it right now.
I guess we can leave it like that for now and come back when there's a need for that?